any_iterator.hpp

```diff
@@ -115,14 +122,23 @@ class any_iterator : std::iterator<std::bidirectional_iterator_tag, T>
     template <typename IterType>
     any_iterator operator=(const IterType& _iter)
     {
+        size_t new_index = getNumber<IterType>();
```

Comment:
Actually this is only safe if T is added to the getNumber parameter list.

Reply:
Anyway, the other code that I'm working on replaces the getNumber/vector thing like this:
```c++
template<class Key, class T>
struct iter_funcs {
  void* (*copy)(const void*);
  void (*assign)(void*, const void*);
  void (*del)(void*);
  bool (*eq)(const void*, const void*);
  void (*inc)(void*);
  void (*dec)(void*);
  std::pair<const Key, T>(*deref)(const void*);
  T(*get_value)(const void*);
  T(*get_key)(const void*);
  void (*set_value)(void*, const T&);
};

template<class Key, class T, class Iter>
iter_funcs<Key,T>* mut_iter_funcs() {
  static iter_funcs<Key,T> funcs{
    wrappers::copy<Iter>,
    wrappers::assign<Iter>,
    wrappers::del<Iter>,
    wrappers::eq<Iter>,
    wrappers::inc<Iter>,
    wrappers::dec<Iter>,
    wrappers::deref<Key,T,Iter>,
    wrappers::get_value<T,Iter>,
    wrappers::get_key<Key,Iter>,
    wrappers::set_value<T,Iter>
  };
  return &funcs;
}
```

Reply:
My code is actually only intended to be used with a relatively small number of well known Iter types, so the assign operation is guaranteed to work in my case.